src/Features/Core/Portable/NavigateTo/NavigateToSearcher.cs

```diff
@@ -28,24 +29,26 @@ internal partial class NavigateToSearcher
         private readonly IImmutableSet<string> _kinds;
         private readonly Document? _currentDocument;
         private readonly IStreamingProgressTracker _progress;
-        private readonly CancellationToken _cancellationToken;
 
-        public NavigateToSearcher(
+        private readonly Document? _activeDocument;
+        private readonly ImmutableArray<Document> _visibleDocuments;
```

Comment:
Is there a need to differentiate the active document from visible documents, or could the active document just be the first item in _visibleDocuments?

Reply:
Yes.  We truly do search for just the _activeDoc first.  It gets *prime* treatment.  Then, we use the _Visible (i.e. open a tab) as a strong hint of what to search next.  And finally, we search anything not hit by that.